app/models/concerns/eligibility.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Eligibility
+  extend ActiveSupport::Concern
+
```

Comment:
This is specific to a user, so the module Eligibility should be scoped as well. Redefine it as `module User::Eligibility`.

Moreover, you should be moving the corresponding validation declarations here as well from the `User::Validations` concern.